payment-processor/src/main/kotlin/org/ostelco/prime/paymentprocessor/StripePaymentProcessor.kt

```diff
@@ -458,10 +513,33 @@ class StripePaymentProcessor : PaymentProcessor {
                 Invoice.create(params)
             }
 
-    override fun payInvoice(invoiceId: String): Either<PaymentError, InvoicePaymentInfo> =
+    /*
+       TODO: Handle payement fails on subscr. creation/renewal.
+
+       Ref. https://stripe.com/docs/billing/subscriptions/payment#handling-failure
+
+       Missing parts:
+          - on create-sub or renewal with status "payment-fails"
+              --  collect invoice-id
+              --  call 'payInvoice' and expand function to include "payment-intent"
+                  in "expand" parameter
+          - if payment fails with 402 - probably fine with java as the invoice is always
+            retrieved first.
+
+       TODO Must have two pay methods, one for subscription renewal and one for all other
+            or maybe not...
+     */
+    override fun payInvoice(invoiceId: String, sourceId: String?): Either<PaymentError, InvoicePaymentInfo> =
             either("Failed to complete payment of invoice ${invoiceId}") {
-                val receipt = Invoice.retrieve(invoiceId).pay()
-                InvoicePaymentInfo(receipt.id, receipt?.charge ?: UUID.randomUUID().toString())
+                val params = if (sourceId != null)
```

Comment:
Please wrap if in parens.  I get confused by this kind of implicitly bounded values.  Be explicit.

Reply:
Agree with that.